### PR TITLE
[ELY-2686] Merge an if statement with the enclosing one in X509EvidenceVerifier

### PR DESCRIPTION
**Pull request by VAIBHAVCES**
[ELY-2686]Merge an if statement with the enclosing one in X509EvidenceVerifier
Jira : https://issues.redhat.com/browse/ELY-2686

### CODE REVIEW

**Comment by PrarthonaPaul** [comment]
Hi @VAIBHAVCES 
Thanks for the PR!
Could you please edit the commit message and the PR description to match the Jira issue and add the link to the Jira here?

**Comment by VAIBHAVCES** [comment]
> Hi @VAIBHAVCES Thanks for the PR! Could you please edit the commit message and the PR description to match the Jira issue and add the link to the Jira here?

@PrarthonaPaul I see  jira issue is already updated the in commit, description and summary.
What else I need to update ?

File: auth/realm/ldap/src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java [hunk 1]
```diff
@@ -217,11 +217,8 @@ public boolean verifyCertificate(X509Certificate certificate, Attributes attribu
             try {
                 for (int i = 0; i < size; i++) {
                     Object attrCertificate = attribute.get(i);
-                    if (attrCertificate != null){
-                        if (MessageDigest.isEqual(certificate.getEncoded(), (byte[]) attrCertificate)) {
-                            return true;
-                        }
-                    }
+                    return attrCertificate != null
```
Comment (fjuma) [inline auth/realm/ldap/src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java]: We need to be a bit careful here. We still need an if statement to make sure we only return true if `attrCertificate != null` and `MessageDigest.isEqual(certificate.getEncoded(), (byte[]) attrCertificate)` are both `true`. If not, the current changes return `false` but that's different from the current behaviour.

**Review by fjuma — CHANGES_REQUESTED** [review]
Added a small comment.

File: auth/realm/ldap/src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java (hunk 1 above)
Comment (VAIBHAVCES) [reply auth/realm/ldap/src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java]: sure reverted @fjuma  please review now

**Comment by PrarthonaPaul** [comment]
> > Hi @VAIBHAVCES Thanks for the PR! Could you please edit the commit message and the PR description to match the Jira issue and add the link to the Jira here?
> 
> @PrarthonaPaul I see jira issue is already updated the in commit, description and summary. What else I need to update ?

Oh, sorry. 
I was referring to the Jira link and updating the commit message to `Merge an if statement with the enclosing one in X509EvidenceVerifier` as opposed to `Removed multiples if and made it single liner`
You can use `git commit --ammend` to do this.

**Comment by VAIBHAVCES** [comment]
> Added a small comment.



> > > Hi @VAIBHAVCES Thanks for the PR! Could you please edit the commit message and the PR description to match the Jira issue and add the link to the Jira here?
> > 
> > 
> > @PrarthonaPaul I see jira issue is already updated the in commit, description and summary. What else I need to update ?
> 
> Oh, sorry. I was referring to the Jira link and updating the commit message to `Merge an if statement with the enclosing one in X509EvidenceVerifier` as opposed to `Removed multiples if and made it single liner` You can use `git commit --ammend` to do this.

done

**Review by PrarthonaPaul — APPROVED** [review]
LGTM!

**Review by fjuma — APPROVED** [review]
Thanks @VAIBHAVCES!

**Review by skyllarr — APPROVED** [review]

**Pull request merged**